sklearn/metrics/_scorer.py

```diff
@@ -711,40 +638,54 @@ def make_scorer(
     scorer : callable
         Callable object that returns a scalar score; greater is better.
 
-    Notes
-    -----
-    If `needs_proba=False` and `needs_threshold=False`, the score
-    function is supposed to accept the output of :term:`predict`. If
-    `needs_proba=True`, the score function is supposed to accept the
-    output of :term:`predict_proba` (For binary `y_true`, the score function is
-    supposed to accept probability of the positive class). If
-    `needs_threshold=True`, the score function is supposed to accept the
-    output of :term:`decision_function` or :term:`predict_proba` when
-    :term:`decision_function` is not present.
-
     Examples
     --------
     >>> from sklearn.metrics import fbeta_score, make_scorer
     >>> ftwo_scorer = make_scorer(fbeta_score, beta=2)
     >>> ftwo_scorer
-    make_scorer(fbeta_score, beta=2)
+    make_scorer(fbeta_score, response_method='predict', beta=2)
     >>> from sklearn.model_selection import GridSearchCV
     >>> from sklearn.svm import LinearSVC
     >>> grid = GridSearchCV(LinearSVC(), param_grid={'C': [1, 10]},
     ...                     scoring=ftwo_scorer)
     """
-    sign = 1 if greater_is_better else -1
-    if needs_proba and needs_threshold:
+    if response_method is not None and (
+        needs_proba != "deprecated" or needs_threshold != "deprecated"
+    ):
         raise ValueError(
-            "Set either needs_proba or needs_threshold to True, but not both."
+            "You cannot set both `response_method` and `needs_proba` or "
+            "`needs_threshold` at the same time. Only use `response_method` since "
+            "the other two are deprecated in version 1.4 and will be removed in 1.6."
+        )
+    if (needs_proba != "deprecated") and (needs_threshold != "deprecated"):
+        raise ValueError(
+            "You cannot set both `needs_proba` and `needs_threshold` at the same "
+            "time. Use `response_method` instead since the other two are deprecated "
+            "in version 1.4 and will be removed in 1.6."
```

Comment:
we should also take into account when they're explicitly passed as `False`

Reply:
So it means that we check both are `True`. The deprecation will be raise later.